app/services/bucket.rb

```diff
@@ -16,14 +16,11 @@ def <<(item)
     @items << item
   end
 
-  def sort_bucket!(field, direction = :asc)
+  def sort_bucket!(field, direction)
     return unless @valid_sort_fields.include?(field)
 
-    if field == :earliest_release_date
-      @items.sort_by! { |e| e.public_send(field) || Date.new(1) }
-    else
-      @items.sort_by!(&field)
-    end
+    # convert nil into '' - date and string sorting should now be handled well
+    @items.sort_by! { |e| e.public_send(field).to_s }
```

Comment:
This change will result in all values being sorted as strings.

### Sorting numbers

This means that numeric values will end up being sorted 'strangely':

```ruby
[2, 100, 10].map(&:to_s).sort
=> ["10", "100", "2"]
```

rather than what a human would typically expect:

```ruby
[2, 100, 10].sort
=> [2, 10, 100]
```

I don't think we currently have any columns with numeric values. It does feel like a bit of a 'UX smell' though – we're replacing one sorting 'quirk' for another – but maybe this is good enough for now? Otherwise I'd suggest we implement [a natural sorting algorithm](https://github.com/rwz/natural_sort).

Reply:
Resolved with the new sorting algorithm in 1bcbb82 – numeric values will now sort.